src/_gettsim_tests/test_lohnst.py

```diff
@@ -0,0 +1,420 @@
+                in Cent""",
+        "BKV": """Bemessungsgrundlage der Vergütung für mehrjährige Tätigkeit
+                  für die Kirchenlohnsteuer in Cent""",
+        "LSTLZZ": "Für den Lohnzahlungszeitraum einzubehaltende Lohnsteuer in Cent",
+        "SOLZLZZ": """Für den Lohnzahlungszeitraum einzubehaltender
+        Solidaritätszuschlag in Cent""",
+        "VKVLZZ": """Für den Lohnzahlungszeitraum berücksichtigte Beiträge des
+        Arbeitnehmers zur privaten Basis Krankenversicherung und privaten Pflege
+        Pflichtversicherung (ggf. auch die Mindestvorsorgepauschale) in Cent beim
+        laufenden Arbeitslohn. Für Zwecke der Lohnsteuerbescheinigung sind die
+        einzelnen Ausgabewerte außerhalb des eigentlichen
+        Lohnsteuerberechnungsprogramms zu addieren;
+         hinzuzurechnen sind auch die Ausgabewerte VKVSONST.""",
+        "VFRB": """Verbrauchter Freibetrag bei Berechnung des
+        laufenden Arbeitslohns, in Cent""",
+    }
+
+    bmf_out = get_bmf_data(url_base, specs, out_definitions)
+    out = bmf_out["value"].astype(int)
+    # Divide by 100 to get Euro
+    return out[outvar] / 100
+
+
+def gen_lohnsteuer_test(year: int, soz_vers_params: dict):
+    """Calls the BMF API to generate correct lohnsteuer payments."""
+
+    hh = pd.DataFrame(
+        {
+            "p_id": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11],
+            "tu_id": [1, 2, 2, 3, 3, 4, 4, 4, 5, 5, 6],
+            "bruttolohn_m": [2000, 3000, 4000, 5000, 0, 2000, 0, 0, 3000, 0, 7500],
+            "alter": [30, 30, 40, 40, 50, 30, 5, 2, 40, 12, 50],
+            "kind": [
+                False,
+                False,
+                False,
+                False,
+                False,
+                False,
+                True,
+                True,
+                False,
+                True,
+                False,
+            ],
+            "steuerklasse": [1, 4, 4, 3, 5, 2, 1, 1, 2, 2, 1],
+            "year": [year] * 11,
+        }
+    )
+    hh["child_num_kg"] = hh.groupby("tu_id")["kind"].transform("sum")
+
+    # Get correct lohnsteuer from German Ministry of Finance
+    hh["lohnst_m"] = np.vectorize(bmf_collect)(
+        inc=hh["bruttolohn_m"],
+        outvar="LSTLZZ",
+        n_kinder=hh["child_num_kg"],
+        stkl=hh["steuerklasse"],
+        jahr=hh["year"],
+        zusatzbeitrag=soz_vers_params["beitr_satz"]["ges_krankenv"][
+            "mean_zusatzbeitrag"
+        ],
+    )
+
+    hh["soli_st_lohnst_m"] = np.vectorize(bmf_collect)(
+        hh["bruttolohn_m"],
+        outvar="SOLZLZZ",
+        n_kinder=hh["child_num_kg"],
+        stkl=hh["steuerklasse"],
+        jahr=hh["year"],
+        zusatzbeitrag=soz_vers_params["beitr_satz"]["ges_krankenv"][
+            "mean_zusatzbeitrag"
+        ],
+    )
+
+    return hh
+
+
+@pytest.mark.skip()
+@pytest.mark.parametrize("year, column", itertools.product([2022], OUT_COLS))
```

Comment:
I think it is a great thing to be able to download stuff directly, but the tests should always be based on a local file.

Please pull out all code calling the API to a module in `src/_gettsim_tests/test_data/original_testfaelle`. When ran, the module should save a file we can use on disk. 

Do I gather correctly that this is where `lohnsteuer_bmf_2022.xlsx` is coming from already?

Reply:
> Do I gather correctly that this is where `lohnsteuer_bmf_2022.xlsx` is coming from already?

No, that file is directly provided by the BMF. And it does not always correspond to the API :sob: Overall, we regard the API as an additional source and hence skip it by default. It looks very attractive at first glance, but it's quite opaque what's happening there and when certain parameters are being updated. It could be two identical API call create different results when being called 6 months apart. Adding a result file to the repository would make things overly complicated in my view.

Reply:
But then we should not keep this code as part of GETTSIM. This always adds a maintenance burden, too. How about a notebook somewhere else, we could link from the docs?

Reply:
Like a notebook in `src/_gettsim_tests`?

Reply:
Honestly I do not see much of a value in keeping it as part of GETTSIM, worrying mostly about a maintenance burden. Dead code is always a drag, no matter what you do (take the current dashboard as a case in point, not sure how much time it cost me over the past three months to properly ignore it everywhere). 

IIUC, what we want to do with it is to call it once a year to check whether the outputs of GETTSIM make sense. We do not need anything internal for that, just the GETTSIM API (sic!). 

Happy to be convinced otherwise, though!

Reply:
I see your point. While these tools (BMF API, Param Dashboard) might be useful in some contexts, the maintenance burden speak for removing them from the 'core' gettsim. What do you think about outsourcing them to additional repositories in the @iza-institute-of-labor-economics organization?

Reply:
Sounds great, will do (though dashboard will probably start over based on plotly/dash instead of bokeh.

Reply:
Here you go: https://github.com/iza-institute-of-labor-economics/gettsim-code-for-picking

Thanks!